src/configs/FirebaseConfig.js

```diff
@@ -189,7 +191,28 @@ const FirebaseConfig = () => {
 
   const handleDeepLinkUser = async (deepLinkUrl) => {
     const username = deepLinkUrl?.split('helio.social/')[1];
-    if (username?.toLocaleLowerCase() === profile?.username?.toLocaleLowerCase()) {
+
+    let profileDataFromStorage;
+    let profileFromApi = null;
+
+    if (!profile?.username) {
+      const profileStorage = StorageUtils.profileData.get();
+      try {
+        profileDataFromStorage = JSON.parse(profileStorage);
+      } catch (e) {
+        console.log('Error parsing profile storage', e);
+      }
+    }
+
+    if (!profileDataFromStorage) {
+      profileFromApi = await getMyProfile();
+      StorageUtils.profileData.set(JSON.stringify(profile.data));
+    }
+
+    const checkedUsername =
+      profile?.username || profileDataFromStorage?.username || profileFromApi?.data?.username || '';
+
+    if (username?.toLocaleLowerCase() === checkedUsername?.toLocaleLowerCase()) {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code introduces a new way to fetch and store profile data for comparison during deep link navigation. It's good that you're handling potential errors when parsing the stored profile data with a try-catch block. However, there seems to be an error on line 209 where you're trying to set the storage with `profile.data` which is not defined in this context. You probably meant to use `profileFromApi.data`.
```diff
- StorageUtils.profileData.set(JSON.stringify(profile.data));
+ StorageUtils.profileData.set(JSON.stringify(profileFromApi.data));
```
Also, consider adding error handling around the `getMyProfile()` API call and the `StorageUtils.profileData.set()` operation as they could potentially fail.


<!-- This is an auto-generated comment by OSS CodeRabbit -->